### PR TITLE
#9281 Implement service sync with no downtime

### PR DESCRIPTION
**Pull request by dionenak**

### CODE REVIEW

**Review by danielbeeke — CHANGES_REQUESTED** [review]

File: src/Service.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { wait } from "./utils/index.js";
 import { _get, _post, _delete, _patch } from "./RequestHandler.js";
 import { getErr, TriplyDbJsError } from "./utils/Error.js";
 import Dataset from "./Dataset.js";
+import msTohms from "./utils/timeHelper.js";
```
[the remark this answers is not in the document]
Comment (dionenak) [reply src/Service.ts]: this is a good question :') . For these reasons, we have a mirror repo where we run pipelines, but I wasn't notified that something went wrong as usual. https://git.triply.cc/triplydb/triplydb-js-mirror/-/pipelines
It's fixed!

[the remark this answers is not in the document]
Comment (dionenak) [reply src/Service.ts]: and I implemented your advice

File: src/Service.ts [hunk 2]
```diff
@@ -162,18 +163,80 @@ export default class Service {
...
   }
 }
+/**
+ * Get the service name string and return a substring,
+ * so that when we make concatinations don't lead to an error
+ * (limit 40 characters for a service name)
+ *
+ * @param serviceName
+ */
+function getSubstrForServiceNames(serviceName: string) {
```
[the remark this answers is not in the document]
Comment (dionenak) [reply src/Service.ts]: done!

**Review by danielbeeke — APPROVED** [review]
Looks good to me

**Pull request merged**